pkg/internal/testhelpers/containers.go

```diff
@@ -130,24 +236,145 @@ func (s stdoutLogConsumer) Accept(l testcontainers.Log) {
 }
 
 // StartPGContainer starts a postgreSQL container for use in testing
-func StartPGContainer(ctx context.Context, usePromscaleExtension bool, useTimescaleDB bool, useTimescale2 bool, testDataDir string, printLogs bool) (testcontainers.Container, error) {
+func StartPGContainer(
+	ctx context.Context,
+	extensionState ExtensionState,
+	testDataDir string,
+	printLogs bool,
+) (testcontainers.Container, io.Closer, error) {
 	var image string
-	if usePromscaleExtension && useTimescale2 {
+	switch extensionState {
+	case MultinodeAndPromscale:
+		image = "timescaledev/promscale-extension:2.0.0-rc2-pg12"
+	case Multinode:
+		image = "timescale/timescaledb:2.0.0-rc2-pg12"
+	case Timescale2AndPromscale:
 		image = "timescaledev/promscale-extension:2.0.0-rc2-pg12"
-	} else if useTimescale2 {
+	case Timescale2:
 		image = "timescale/timescaledb:2.0.0-rc2-pg12"
-	} else if usePromscaleExtension && useTimescaleDB {
+	case Timescale1AndPromscale:
 		image = "timescaledev/promscale-extension:latest-pg12"
-	} else if useTimescaleDB {
+	case Timescale1:
 		image = "timescale/timescaledb:latest-pg12"
-	} else {
+	case VanillaPostgres:
 		image = "postgres:12"
 	}
-	return StartPGContainerWithImage(ctx, image, testDataDir, "", printLogs, usePromscaleExtension, useTimescaleDB)
+
+	return StartPGContainerWithImage(ctx, image, testDataDir, "", printLogs, extensionState)
 }
 
-func StartPGContainerWithImage(ctx context.Context, image string, testDataDir string, dataDir string, printLogs bool, usePromscaleExtension bool, useTimescaleDB bool) (testcontainers.Container, error) {
+func StartPGContainerWithImage(ctx context.Context,
+	image string,
+	testDataDir string,
+	dataDir string,
+	printLogs bool,
+	extensionState ExtensionState,
+) (testcontainers.Container, io.Closer, error) {
+
+	multinode = extensionState.usesMultinode()
+
+	c := CloseAll{}
+	var networks []string
+	if multinode {
+		networkName, closer, err := createMultinodeNetwork()
+		if err != nil {
+			return nil, c, err
+		}
+		c.Append(closer)
+		networks = []string{networkName}
+	}
+
+	startContainer := func(containerPort nat.Port) (testcontainers.Container, error) {
+		container, closer, err := startPGInstance(
+			ctx,
+			image,
+			testDataDir,
+			dataDir,
+			extensionState,
+			printLogs,
+			networks,
+			containerPort,
+		)
+		if closer != nil {
+			c.Append(closer)
+		}
+		return container, err
+	}
+
+	if multinode {
+		containerPort := nat.Port("5433/tcp")
+		_, err := startContainer(containerPort)
+		if err != nil {
+			_ = c.Close()
+			return nil, nil, err
+		}
+		containerPort = nat.Port("5434/tcp")
+		_, err = startContainer(containerPort)
+		if err != nil {
+			_ = c.Close()
+			return nil, nil, err
+		}
+	}
+
 	containerPort := nat.Port("5432/tcp")
+	container, err := startContainer(containerPort)
+	if err != nil {
+		_ = c.Close()
+		return nil, nil, err
+	}
+
+	if multinode {
+		db, err := pgx.Connect(context.Background(), PgConnectURL(defaultDB, Superuser))
```

Comment:
why not do the connect and shutdown within enableDataNodes if connection never used anywhere else?

Reply:
I want to use the same function in [`DbSetup`](https://github.com/timescale/promscale/pull/334/files#diff-a9a488c67e814258e1b5878fa9c9160b6b011e45d10772a918341c983d185204R198) and there the connection is used for other things (not married to that, I can move the connection into the other branch as well)